File: lims-web-app/src/services/MasterDataService/ReportSettingsMaster/ReportPaperSizeService.js

import { Urls } from '../../apiURLS'
import axios from 'axios';
const BASE_URL=Urls.MASTERS;
export const getAllReportPaperSizes = async (startsWith = '', pageNumber = 0, pageSize = 10, sortBy = 'paperSize', createdBy = '') => {
    const response = await axios.post(`${BASE_URL}report-paper-size/get-all?startsWith=${startsWith}&pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=${sortBy}`, {}, { headers: { 'createdBy': createdBy } });
    return response.data;
};

export const addReportPaperSize = async (reportPaperSize, createdBy) => {
    const response = await axios.post(`${BASE_URL}report-paper-size/save`, reportPaperSize, { headers: { 'createdBy': createdBy } });
    return response.data;
};
export const deleteReportPaperSizeById= async(id)=>{
    const response= await axios.delete(`${BASE_URL}report-paper-size/${id}`);
    return response.data;
}
export const getReportPaperSizeById=async(id)=>{
    const response= await axios.get(`${BASE_URL}report-paper-size/${id}`);
    return response.data;
}
export const updateReportPaperSize=async(id, updatedReportPaperSize, userId) => {
    const response=await axios.put(`${BASE_URL}report-paper-size/update/${id}`, updatedReportPaperSize, { headers: { 'userId': userId } });
    return response.data;
}
export const reportPaperSizeService = {
    getAllReportPaperSizes,
    addReportPaperSize,
    deleteReportPaperSizeById,
    getReportPaperSizeById,
    updateReportPaperSize
};

File: back-end-lims/master-data-service/src/main/java/com/digiworldexpo/lims/master/service/ReportPaperSizeService.java

package com.digiworldexpo.lims.master.service;



import com.digiworldexpo.lims.entities.master.ReportPaperSize;
import com.digiworldexpo.lims.master.model.response.ResponseModel;

import java.util.List;
import java.util.UUID;

public interface ReportPaperSizeService {

	ResponseModel<ReportPaperSize> saveReportPaperSize(ReportPaperSize reportPaperSize, UUID createdBy);

	ResponseModel<List<ReportPaperSize>> getReportPaperSizes(String startsWith, int pageNumber, int pageSize,
			String sortBy, UUID createdBy);

	ResponseModel<ReportPaperSize> getReportPaperSizeById(UUID id);

	ResponseModel<ReportPaperSize> updateReportPaperSize(UUID id, ReportPaperSize updatedReportPaperSize, UUID userId);

	ResponseModel<ReportPaperSize> deleteReportPaperSize(UUID id);
}
